src/src/TestGroup.php

```diff
@@ -535,4 +540,35 @@ public function fetch( string $group_identifier ): array {
 
 		return json_decode( $response, true );
 	}
+
+	public function output_group( OutputInterface $output ): void {
+		$group = $this->get();
+		if ( ! array_key_exists( 'group_identifier', $group ) ) {
```

Comment:
Should we ensure that $group is an array here in order to make it more robust? Something like:
```
if ( is_array( $group ) && ! array_key_exists( 'group_identifier', $group ) )
```

Reply:
We don't need to. The function `get` method on the TestGroup class returns an empty array if the group doesn't exist and everywhere else where we fetch the group directly from the cache has defensive checks in place